tensorflow/python/ops/math_ops.py

```diff
@@ -966,6 +966,12 @@ def cast(x, dtype, name=None):
 
   Note casting nan and inf values to integral types has undefined behavior.
 
+  Note this operation can lead to a loss of precision when converting non-tensor
+  inputs such as native Python `float` and `complex` variables to `tf.float64`
```

Comment:
I think we need restrict to python types only:
```
Note this operation can lead to a loss of precision when converting native Python `float` and `complex` types...
```
Because as you noticed earlier, `np.float64` arrays are non-tensor inputs but do not lose precision.